### PR TITLE
fix(rnd): Fix execution error on non-saved agent

### PR DESCRIPTION
**Pull request by majdyz**
### Background

https://github.com/Significant-Gravitas/AutoGPT/pull/8052#discussion_r1759764573
Introduced a new execution bug in which the node that hasn't been saved is unable to execute.

image

### Changes 🏗️

The change reverts the change introduced in the mentioned PR and does it differently to avoid impacting the saved node id.
And fix the behaviour of running an agent that is not saved yet by disabling it.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Comment by Ujjawal-Anand** [comment]
The new agent flow still isn't working as expected. Currently, when a user clicks "Run" for new agents, we save the agent and then trigger a router change. This leads to a reset of all states, including the `saveRunRequest` state, which is set to default `none`. In this case, while the agent does get executed on the server, the result is not reflected on the frontend due to the state reset.

Issue 2: When I clicked "Run" again, the agent did not execute. I noticed that the `saveRunRequest` state remained `{request: run, state: saving}` indefinitely.

Note that i didn’t observed any issues for my previously saved agents. 
The issue i mentioned, is only happening for new unsaved and saved agent.

**Review by daki90 — APPROVED** [review]

**Comment by majdyz** [comment]
@Ujjawal-Anand yeah this is tricky. Running an unsaved agent is a corner case by itself. To run an agent you need to save it, and previously we will just save it with a random name and description. It will still show up with that default name on the monitoring page. In this PR I'm removing that possibility entirely by simply disabling the run button on the unsaved graph.

cc @kcze

File: rnd/autogpt_builder/src/components/edit/control/ControlPanel.tsx [hunk 1]
```diff
@@ -50,15 +51,18 @@ export const ControlPanel = ({
...
+                    variant="ghost"
+                    size="icon"
+                    onClick={() => control.onClick()}
+                    data-id={`control-button-${index}`}
+                    disabled={control.disabled || false}
+                  >
+                    {control.icon}
+                    <span className="sr-only">{control.label}</span>
+                  </Button>
+                </div>
```
Comment (majdyz) [inline rnd/autogpt_builder/src/components/edit/control/ControlPanel.tsx:65]: A tooltip won't show up on disabled button unless I do this.

File: rnd/autogpt_builder/src/components/ui/input.tsx [hunk 2]
```diff
@@ -6,20 +6,7 @@ export interface InputProps
   extends React.InputHTMLAttributes<HTMLInputElement> {}
 
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type, value, ...props }, ref) => {
```
Comment (majdyz) [inline rnd/autogpt_builder/src/components/ui/input.tsx:9]: Let's just make an Input be input, this simplifies a lot.

**Review by Swiftyos — APPROVED** [review]

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**